src/uct/api/uct.h

```diff
@@ -1162,9 +1203,57 @@ struct uct_ep_params {
      * 0..(@ref uct_iface_attr_t.dev_num_paths - 1).
      */
     unsigned                            path_index;
+
+    /**
+     * Callback is called when the client side endpoint is bound to network
+     * device and provides related info as defined in
+     * @ref uct_cm_ep_resolve_args_t. This field is only valid if
+     * @ref uct_ep_params_t::sockaddr is set.
+     * @note It is never guaranteed that the callaback will be called. If, for
+     * example, the endpoint goes into error state before issuing the connection
+     * request, the callback will not be invoked.
+     */
+    uct_cm_ep_resolve_callback_t        sockaddr_resolve_cb;
+
+    /**
+     * User's private data to be passed from server to client, can be
+     * initialized only together with @a conn_request, can not be initialized
+     * together with @a sockaddr_pack_cb.
```

Comment:
Private data to be passed from server to client. Can be used only along with `@ref uct_ep_params.conn_request`.
note
This field is mutually exclusive with `@ref uct_ep_params.XYZ`